pkg/api/response/response.go

```diff
@@ -214,6 +219,29 @@ func Error(status int, message string, err error) *NormalResponse {
 	return resp
 }
 
+// Err creates an error response based on an errutil.Error error.
+func Err(log log.Logger, err error) *NormalResponse {
+	grafanaErr := &errutil.Error{}
+	if !errors.As(err, grafanaErr) {
+		log.Error(
+			"unexpected error type",
+			"err", err,
+			"type", reflect.TypeOf(err),
+		)
+		return Error(http.StatusInternalServerError, "", err)
+	}
+
+	grafanaErr.Log(log)
```

Comment:
we not logging always in `Error`? I think [that's what is happening now](https://github.com/grafana/grafana/blob/4fbd974471b62a985e6c923d07ac0a39e69821e0/pkg/api/response/response.go#L85)

Reply:
No, only if we set the error for the response, and that logs every `error` with severity "error". The `errutil.Error` type is meant to be used for things like 404s and not just 500s, which definitely shouldn't be logged quite that loudly.

However, it is a problem that it'd be logged twice in such a case, I had missed that we had that logger. I'll have to ponder a bit on how to structure this.

Reply:
Fixed this